optuna/integration/allennlp/_pruner.py

```diff
@@ -165,11 +162,22 @@ def __init__(
         # `trial` and `monitor` would be None. `AllenNLPExecutor` sets information
         # for a study name, trial id, monitor, and storage in environment variables.
         else:
-            environment_variables = _get_environment_variables_for_trial()
-            study_name = environment_variables["study_name"]
-            trial_id = environment_variables["trial_id"]
-            monitor = environment_variables["monitor"]
-            storage = environment_variables["storage"]
+            current_process = psutil.Process()
+
+            if os.getenv("OPTUNA_ALLENNLP_USE_DISTRIBUTED") == "1":
+                os.environ.pop("OPTUNA_ALLENNLP_USE_DISTRIBUTED")
+                parent_process = current_process.parent()
+                target_pid = parent_process.ppid()
```

Comment:
Here, I get an id of a grandparent process. This is because pruner initialization is executed on child processes spawned by `torch.multiprocessing` in distributed training.